src/main/java/org/spongepowered/api/text/dialogue/AdditionalAnswerProcessor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.dialogue;
+
+/**
+ * A functional interface that processes answers without deciding the next
+ * {@link Question}.
+ */
+@FunctionalInterface
+public interface AdditionalAnswerProcessor {
+
+    /**
+     * Processes a {@link Question}.
+     *
+     * @param dialogue The current dialogue
+     * @param question The question being answered
+     * @param answer The answer to the question
+     */
+    void process(Dialogue dialogue, Question question, Answer answer);
```

Comment:
I feel like providing the dialogue and Question here is a little redundant and useless.  Answer exists as a container type containing both the Dialogue and Question, and is referenced only in the AnswerProcessors.  AdditionalAnswerProcessor could be replaced by a Consumer<Answer>, and AnswerProcessor by a Function<Answer, Optional<Question>>.

You mentioned before that you provide them for convenience, but I find that excessive arguments is generally an inconvenience when it comes to functional interfaces.  If you wanted to use a method reference as an AnswerProcessor, you'd need every one of your methods to accept all three arguments, rather than just the one actually needed argument.  You're limited purely to your own implementations of AnswerProcessor.  By providing only the Answer, you allow interaction with anything that's a generic Consumer.